src/main/java/com/openlattice/chronicle/controllers/ChronicleStudyController.java

```diff
@@ -111,238 +106,357 @@ public Boolean isKnownDatasource(
         //  validate that this device belongs to this participant in this study
         //  look up in association entitySet between device and participant, and device and study to see if it exists?
         //  DataApi.getEntity(entitySetId :UUID, entityKeyId :UUID)
-        // TODO: Waiting on data model to exist, then ready to implement
-        return chronicleService.isKnownDatasource( studyId, participantId, datasourceId );
+        return chronicleService.isKnownDatasource( CAFE_ORG_ID, studyId, participantId, datasourceId );
+    }
+
+    @Override
+    @RequestMapping(
+            path = ORGANIZATION_ID_PATH + STUDY_ID_PATH + PARTICIPANT_ID_PATH + DATASOURCE_ID_PATH + VALID,
+            method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE )
+    public Boolean isKnownOrgStudyDataSource(
+            @PathVariable( ORGANIZATION_ID ) UUID organizationId,
+            @PathVariable( STUDY_ID ) UUID studyId,
+            @PathVariable( PARTICIPANT_ID ) String participantId,
+            @PathVariable( DATASOURCE_ID ) String datasourceId ) {
+        return chronicleService.isKnownDatasource( organizationId, studyId, participantId, datasourceId );
```

Comment:
I agreed. In addition, we can safely remove `isKnownDatasource` usage from the chronicle app since`enrollSource` will return the device EKID if the device is already associated with participant.